code/lib/cli/src/automigrate/fixes/sveltekit-framework.test.ts

```diff
@@ -0,0 +1,173 @@
+import type { StorybookConfig } from 'lib/types/src';
+import path from 'path';
+import type { JsPackageManager, PackageJson } from '../../js-package-manager';
+import { sveltekitFramework } from './sveltekit-framework';
+
+// eslint-disable-next-line global-require, jest/no-mocks-import
+jest.mock('fs-extra', () => require('../../../../../__mocks__/fs-extra'));
+
+const checkSvelteKitFramework = async ({
+  packageJson,
+  main,
+}: {
+  packageJson: PackageJson;
+  main?: Partial<StorybookConfig>;
+}) => {
+  if (main) {
+    // eslint-disable-next-line global-require
+    require('fs-extra').__setMockFiles({
+      [path.join('.storybook', 'main.js')]: `module.exports = ${JSON.stringify(main)};`,
+    });
+  } else {
+    // eslint-disable-next-line global-require
+    require('fs-extra').__setMockFiles({});
+  }
+  const packageManager = {
+    retrievePackageJson: () => ({ dependencies: {}, devDependencies: {}, ...packageJson }),
+  } as JsPackageManager;
+
+  return sveltekitFramework.check({ packageManager });
+};
+
+describe('SvelteKit framework fix', () => {
+  describe('should no-op', () => {
+    it('in SB < v7.0.0', async () => {
+      const packageJson = {
+        dependencies: { '@sveltejs/kit': '^1.0.0-next.571', '@storybook/svelte': '^6.2.0' },
+      };
+      const main = { framework: '@storybook/svelte-vite' };
+      await expect(checkSvelteKitFramework({ packageJson, main })).resolves.toBeFalsy();
+    });
+
+    describe('in SB > v7.0.0', () => {
+      it('in non-SvelteKit projects', async () => {
+        const packageJson = {
+          dependencies: { svelte: '^3.53.1', '@storybook/svelte-vite': '^7.0.0' },
+        };
+        const main = {
+          framework: '@storybook/svelte-vite',
+        };
+        await expect(checkSvelteKitFramework({ packageJson, main })).resolves.toBeFalsy();
+      });
+
+      it('without main', async () => {
+        const packageJson = {
+          dependencies: { '@sveltejs/kit': '^1.0.0-next.571', '@storybook/svelte': '^7.0.0' },
+        };
+        await expect(checkSvelteKitFramework({ packageJson })).rejects.toThrow();
+      });
+
+      it('without framework field in main', async () => {
+        const packageJson = {
+          dependencies: { '@sveltejs/kit': '^1.0.0-next.571', '@storybook/svelte': '^7.0.0' },
+        };
+        const main = {};
+        await expect(checkSvelteKitFramework({ packageJson, main })).rejects.toThrow();
+      });
+
+      it('with unsupported framework', async () => {
+        const packageJson = {
+          dependencies: {
+            '@sveltejs/kit': '^1.0.0-next.571',
+            '@storybook/svelte-vite': '^7.0.0',
+            '@storybook/html': '^7.0.0',
+          },
+        };
+        const main = {
+          framework: '@storybook/html',
+        };
+        await expect(checkSvelteKitFramework({ packageJson, main })).rejects.toThrow();
+      });
+
+      it('with unsupported framework+builder from SB 6.5', async () => {
```

Comment:
A little surprising. Don't we want to suggest for them to update to the sveltekit framework?

Reply:
This is the warning we're showing them when they're using the webpack builder:
https://github.com/storybookjs/storybook/pull/20094/files#diff-7b86c6a0f4de84b22a855dcfd9f725b5171990e63366498f77289606869de4ccR106-R113

Don't you think that's sufficient?

It's very weird if users are using Svelte+Webpack in their SvelteKit project, but our problem is that we can't migrate any of their custom `webpackFinal` stuff, so we can't automigrate it (at least not without a lot of work).